training/benchmarks/glm/pytorch/schedulers/base.py

```diff
@@ -54,8 +54,8 @@ def get_lr(self):
             return float(self.start_lr) * self.num_iters / self.warmup_iter
         else:
             if self.decay_style == self.DECAY_STYLES[0]:
-                decay_step_ratio = (self.num_iters -
-                                    self.warmup_iter) / self.end_iter
+                decay_step_ratio = min(
```

Comment:
prevent learning_rate from becoming negative

Reply:
it works